bash/create-service-principal-2.0.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
```

Comment:
Huh didn't realize the 2.0 cli is generally available as of last month. In any case, I think we should *make* the original script the 2.0 version. What is the benefit of two separate scripts?

I put together a quick example of what I'm talking about. I just copied your commands and I didn't test it out, but it uses less copied code and uses the 2.0 cli by default: https://github.com/Azure/azure-devops-utils/commit/39f905deb0272827bcca9921e4f966cf9cb9006e